server.js

```diff
@@ -1,16 +1,121 @@
 const express = require('express');
 const helmet = require('helmet');
 const cors = require('cors');
+const mongoose = require('mongoose');
 
 const server = express();
 
 server.use(helmet());
 server.use(cors());
 server.use(express.json());
 
+mongoose
+  .connect('mongodb://localhost/Friends')
+  .then((connect) => {
+    console.log('API Server Connected');
+  })
+  .catch((error) => {
+    console.error('Db connection failed');
+  });
+
 server.get('/', (req, res) => {
   res.status(200).json({ api: 'running' });
 });
 
+server.post('/api/friends', (req, res) => {
+  const { firstName, lastName, age, createdOn } = req.body;
+  createdOn = !createdOn ? new Date() : createdOn; // initialize createdOn to current date/time if it is undefined
+  if (!firstName || !lastName || !age) {
+    res.status(400).json({ errorMessage: 'Please provide firstName, lastName, and age for the friend.' });
```

Comment:
I love that you're validating user input! 👌Great job with that!

Make sure you put a `return` inside the `if` statement though so our function's execution ends immediately. Otherwise the _rest_ of our function will execute sending _multiple_ `res.status().json()`s to the client!

```
  if (!firstName || !lastName || !age) {
    return res.status(400).json({ errorMessage: 'Please provide firstName, lastName, and age for the friend.' });
  }
  if (typeof age !== 'number' || age < 1 && age > 120) {
    return res.status(400).json({ errorMessage: 'Age must be a number between 1 and 120.' });
  }
```